### PR TITLE
Session management

### PR DESCRIPTION
**Pull request by joseleonardorojas**
## Summary

* Add book list view
* Routes
* Session management

problems
* typing of function parameters
* routes

## Screenshots

image

## Trello Card

https://trello.com/c/5hYMLKd2/15-session-management-logout
https://trello.com/c/fWYHxOxk/14-book-list-auth-header

### CODE REVIEW

File: src/components/App/index.tsx [hunk 1]
```diff
@@ -15,8 +19,30 @@ function App() {
...
+            }
+          />
+          <Route
+            path="/"
+            element={
+              <PublicRoute>
+                <Login />
+              </PublicRoute>
+            }
+          />
```
Comment (renzowolox) [inline src/components/App/index.tsx]:
```suggestion
          <Route element={<PublicRoute />}>
            <Route path="/sign_up" element={<SignUp />}/>
            <Route path="/" element={<Login />}/>
          </Route>
```

File: src/components/BookList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+function NavBar() {
+  const { data, isLoading } = useQuery('book', getBooks);
+  const books = data?.page || [];
+  return (
+    <div className={styles.container}>
+      {isLoading ? (
+        <Loading />
+      ) : (
+        books.map((todo: IBook) => (
```
Comment (renzowolox) [inline src/components/BookList/index.tsx]:
```suggestion
        books.map((book: IBook) => (
```
Use more descriptive names

File: src/components/BookList/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+.img-book {
+  height: 200px;
+  width: 142px;
+}
+
+.title-book {
+  color: $black;
+  font-family: Roboto;
+  font-weight: 700;
+  font-size: 16px;
```
Comment (renzowolox) [inline src/components/BookList/styles.module.scss]: Sort

File: src/components/BookList/styles.module.scss [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+.title-book {
+  color: $black;
+  font-family: Roboto;
+  font-weight: 700;
+  font-size: 16px;  
+  line-height: 18px;
+}
+
+.author-book {
+  color: #828282;
```
Comment (renzowolox) [inline src/components/BookList/styles.module.scss]: use color variable

File: src/components/NavBar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  const logout = () => {
+    LocalStorageService.removeValue('access-token');
+    navigate('/', { replace: true });
+  };
+
+  return (
+    <div className={styles.navBar}>
+      <img className={styles.logo} src={logo} />
+      <button type="button" onClick={logout} className={styles.btnLogout}>
+        Logout
```
Comment (renzowolox) [inline src/components/NavBar/index.tsx]: Add i18n

File: src/components/Notification/index.tsx [hunk 6]
```diff
@@ -11,4 +11,4 @@ function Notification({ message, type }: Props) {
   return <div className={type === 'error' ? styles.msgError : styles.msgSuccess}>{ message }</div>;
 }
 
-export default Notification;
+export default Notification;
```
Comment (renzowolox) [inline src/components/Notification/index.tsx]: Add empty line

File: src/components/PrivateRoute/index.tsx [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { Navigate } from 'react-router-dom';
+
+import { isAuthenticated } from 'utils/sessionManagement';
+
+function Notification({ children }: any) {
```
Comment (renzowolox) [inline src/components/PrivateRoute/index.tsx]: Why is it called Notification?

File: src/components/PublicRoutes/index.tsx [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { Navigate } from 'react-router-dom';
+
+import { isAuthenticated } from 'utils/sessionManagement';
+
+function Notification({ children }: any) {
+  return isAuthenticated() ? <Navigate to="/home" /> : children;
+}
+
+export default Notification;
```
Comment (renzowolox) [inline src/components/PublicRoutes/index.tsx]:
```suggestion
import React from 'react';
import { Navigate, Outlet } from 'react-router-dom';

import { isAuthenticated } from 'utils/sessionManagement';

function PublicRoutes() {
  if (isAuthenticated()) {
    return <Navigate to="/home" replace />;
  }
  
  return <Outlet />;
}

export default PublicRoutes;
```

File: src/screens/SignUp/index.tsx [hunk 9]
```diff
@@ -10,7 +10,7 @@ import Input from '../../components/Input/index';
 import { validations } from '../../constants/formsValidation';
 import { signUp } from '../../services/UsersService';
 import Loading from '../../components/Spinner/components/loading';
-import Notification from '../../components/Notification/index';
+import Notification from '../../components/PrivateRoute/index';
```
Comment (renzowolox) [inline src/screens/SignUp/index.tsx]: why replaced Notification with PrivateRoute component?

**Review by renzowolox — CHANGES_REQUESTED** [review]

File: src/components/App/index.tsx [hunk 10]
```diff
@@ -15,8 +19,13 @@ function App() {
...
-          <Route path="/sign_up" element={<SignUp />} />
-          <Route path="/" element={<Login />} />
+          <Route element={<PrivateRoute />}>
+            <Route path="/home" element={<Home />} />
+          </Route>
+          <Route element={<PublicRoute />}>
+            <Route path="/sign_up" element={<SignUp />} />
+            <Route path="/" element={<Login />} />
+          </Route>
           <Route path="*" element={<div>No found</div>} />
```
Comment (betoobernal) [inline src/components/App/index.tsx]: use i18n

File: src/components/BookList/styles.module.scss [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+.container {
+  display: grid;
+  gap: 36px;
+  grid-template-columns: repeat(4, auto);
+  justify-content: center; 
+  margin: 45px 0; 
+}
+
+.cardBook {
```
Comment (betoobernal) [inline src/components/BookList/styles.module.scss]: use kebab-case

File: src/components/NavBar/index.tsx [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
+import styles from './styles.module.scss';
+
+function NavBar() {
+  const navigate = useNavigate();
+  const { t } = useTranslation();
+
+  const logout = () => {
+    LocalStorageService.removeValue('access-token');
+    navigate('/', { replace: true });
```
Comment (betoobernal) [inline src/components/NavBar/index.tsx]: i suppose we could use a constant file for this

**Pull request merged**